src/Zicht/Bundle/PageBundle/Security/Voter/ScheduledContentVoter.php

```diff
@@ -21,46 +21,72 @@ class ScheduledContentVoter extends AdminAwareVoterAbstract
      * Decide based on the current date and time what the vote should be. Static so it's strategy can easily be accessed
      * by other components as well, without the actual need for the voter instance.
      *
-     * @param \DateTime $scheduledFrom
-     * @param \DateTime $scheduledTill
+     * @param ScheduledContentInterface $object
+     * @param array $attributes
      * @return int
      */
-    public static function decide(ScheduledContentInterface $object)
+    public static function decide(ScheduledContentInterface $object, array $attributes = [])
     {
-        $scheduledFrom = $object->isScheduledFrom();
-        $scheduledTill = $object->isScheduledTill();
-
+        $from = $object->isScheduledFrom();
+        $till = $object->isScheduledTill();
+        $now = new \DateTimeImmutable();
         $vote = VoterInterface::ACCESS_ABSTAIN;
 
-        if (null !== $scheduledFrom || null !== $scheduledTill) {
-            $currentDateTime = new \DateTime();
-            if (null !== $scheduledFrom && null !== $scheduledTill) {
-                // Check between datetime objects
-                $vote = $scheduledFrom <= $currentDateTime && $scheduledTill >= $currentDateTime ? VoterInterface::ACCESS_GRANTED : VoterInterface::ACCESS_DENIED;
-            } elseif (null !== $scheduledFrom && null === $scheduledTill) {
+        if (!$object->isPublic()) {
+            return $vote;
+        }
+
+        if (null !== $from || null !== $till) {
```

Comment:
je zou liever hier de inverse van nemen en gelijk eruit kunnen stappen. want deze if statement word hieronder ook al afgevangen in verschillende scenarios.

en zou je meer comments erin willen zetten wat je doet want het is best een lang stuk code en ik heb er moeite mee om het te begrijpen